src/com/goide/refactor/GoIntroduceVariableBase.java

```diff
@@ -0,0 +1,252 @@
+    PsiRecursiveElementVisitor visitor = new PsiRecursiveElementVisitor() {
+      public void visitElement(@NotNull PsiElement element) {
+        if (PsiEquivalenceUtil.areElementsEquivalent(element, pattern)) {
+          occurrences.add(element);
+          return;
+        }
+        super.visitElement(element);
+      }
+    };
+    context.acceptChildren(visitor);
+    return occurrences;
+  }
+
+  protected static void performOnElement(@NotNull final Project project,
+                                         @NotNull final Editor editor,
+                                         @NotNull final GoExpression expression) {
+    if (GoInspectionUtil.getExpressionResultCount(expression) != 1) {
+      showCannotPerform(project, editor, "Expression " + expression.getText() + " returns multiple values", null);
+      return;
+    }
+    PsiElement statement = PsiTreeUtil.getParentOfType(expression, GoStatement.class);
+    if (statement == null) {
+      showCannotPerform(project, editor);
+      return;
+    }
+    final List<PsiElement> occurrences = getOccurrences(expression, PsiTreeUtil.getParentOfType(statement, GoBlock.class));
+    OccurrencesChooser.simpleChooser(editor).showChooser(expression, occurrences, new Pass<OccurrencesChooser.ReplaceChoice>() {
+      @Override
+      public void pass(OccurrencesChooser.ReplaceChoice choice) {
+        performOnElementOccurrences(project, editor, expression, occurrences, choice == OccurrencesChooser.ReplaceChoice.ALL);
+      }
+    });
+  }
+
+  private static boolean isCommonAncestor(@NotNull PsiElement ancestor, @NotNull List<PsiElement> occurrences) {
+    for (PsiElement element : occurrences) {
+      if (!PsiTreeUtil.isAncestor(ancestor, element, false)) return false;
+    }
+    return true;
+  }
+
+  private static PsiElement findAnchor(@NotNull List<PsiElement> occurrences) {
+    PsiElement statement = ContainerUtil.getFirstItem(occurrences);
+    while (true) {
+      statement = PsiTreeUtil.getParentOfType(statement, GoStatement.class);
+      if (statement == null) return null;
+      if (isCommonAncestor(statement.getParent(), occurrences)) return statement;
+    }
+  }
+
+  private static void performOnElementOccurrences(@NotNull final Project project,
+                                                  @NotNull final Editor editor,
+                                                  @NotNull final GoExpression expression,
+                                                  final List<PsiElement> occurrences, final boolean replaceAll) {
+    final PsiElement anchor = replaceAll ? findAnchor(occurrences) : PsiTreeUtil.getParentOfType(expression, GoStatement.class);
+    if (anchor == null) {
+      showCannotPerform(project, editor, "Cannot Perform Refactoring in Current Context", null);
+      return;
+    }
+    LinkedHashSet<String> suggestedNames = getSuggestedNames(expression);
+    final String name = ContainerUtil.getFirstItem(suggestedNames);
+    assert name != null;
```

Comment:
yet another set of nulls, is it even possible and will performInplaceRefactoring work with set of nulls?